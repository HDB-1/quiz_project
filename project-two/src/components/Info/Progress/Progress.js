import React from 'react'

export default function Progress(props) {
    return (
        <div>
            <h5 id="progress">{props.question.current +1}/{props.question.total}</h5>
        </div>
    )
}
