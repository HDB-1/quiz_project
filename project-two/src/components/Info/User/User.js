import React from 'react'

export default function User(props) {
    return (
        <div>
            <p id="currentUser">Player {props.currentUser + 1}</p>
        </div>
    )
}
