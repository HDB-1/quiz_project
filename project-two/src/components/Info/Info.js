import React from 'react'
import Title from './Title/Title'
import Progress from './Progress/Progress'
import User from './User/User'
export default function Info(props) {
    return (
        <div>
            <User currentUser={props.users}/>
            <Title title={props.title}/>
            <Progress question={{current:1, total:10}}/>
        </div>
    )
}
