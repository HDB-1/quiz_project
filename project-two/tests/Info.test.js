import React from 'react';
import { shallow , mount } from 'enzyme';
import Info from '../src/components/Info/Info';
import Progress from '../src/components/Info/Progress/Progress'
import User from '../src/components/Info/User/User'

describe('Shallow Info', () => {
  let wrapper;
  
  beforeEach(() => wrapper = shallow(<Info title="General Knowledge" />)); // this automatically tests for rendering without crashing.
  it("Should render correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("Should render 1 <Progress/> and <User/> component and 1 element with id 'quizName'", () =>{
    expect(wrapper.find(Progress).length).toEqual(1);
    expect(wrapper.find(User).length).toEqual(1);
    expect(wrapper.find('#quizName').length).toEqual(1);
  });
  it("Should render information passed to title, progress, user components correctly", () => {
    wrapper = mount(<Info users={1} question={{current:1, total:10}}/>);
    expect(wrapper.find(User).find('#currentUser').text()).toEqual("Player 1");
    expect(wrapper.find(Progress).find("#progress").text()).toEqual("1/10");
  });
  it("The element with ID quizName should render the correct information", () => {
    expect(wrapper.find('#quizName').text()).toEqual('General Knowledge');
  });
});