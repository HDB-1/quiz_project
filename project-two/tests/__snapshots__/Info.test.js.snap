// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shallow Info Should render correctly 1`] = `
<div>
  <User />
  <Title />
  <Progress
    question={
      Object {
        "current": 1,
        "total": 10,
      }
    }
  />
</div>
`;
